### DISCUSSION TITLE
Issue with specific bundle component are not loaded in website.

### DISCUSSION THREAD

**Discussion by Paresh-Dhanani** [breezefront/community#217] [1. Help] [answered]
I am using breeze theme with multi website and different theme for desktop and mobile. Now i have some component which are loading sometime in desktop or mobile and sometime not and when i checked default.min.js then i can't found my component code inside. Please help me here as i have also tried by define different component name for mobile and desktop but still i am facing issue.

**Comment by vovayatsyuk** [comment]
What XML file and what code do you use to add js files to breeze bundles?

**Comment by Paresh-Dhanani** [comment]
Please find the code which i am using for mobile and desktop theme.

xml file

```
         <referenceBlock name="breeze.js">
            <arguments>
                <argument name="bundles" xsi:type="array">
                    <item name="default" xsi:type="array">
                        <item name="items" xsi:type="array">
                            <item name="CP_Homepagereview/js/personalization" xsi:type="array">
                                <item name="path" xsi:type="string">CP_Homepagereview/js/breeze/personalization</item>
                                <item name="import" xsi:type="array">
                                    <item name="priceUtils" xsi:type="string">Swissup_Breeze/js/common/price-utils</item>
                                </item>
                            </item>
                        </item>
                    </item>
                </argument>
            </arguments>
        </referenceBlock>
```

js file

```
$.widget('mage.homepage_personalize', {
            component: 'CP_Homepagereview/js/personalization',
            options: {
             }
}
```

**Reply by vovayatsyuk** [reply]
What is the path to your xml file?

**Comment by Paresh-Dhanani** [comment]
Here I have 2 themes for desktop and mobile and the path is as per below. It is coming sometimes on the page and sometimes not.

app/design/frontend/Test/theme1/CP_Homepagereview/layout/breeze_cms_index_index.xml
app/design/frontend/Test/theme1/CP_Homepagereview/templates/homepersonalize.phtml
app/design/frontend/Test/theme1/CP_Homepagereview/web/js/breeze/personalization.js

app/design/frontend/Test/theme2/CP_Homepagereview/layout/breeze_cms_index_index.xml
app/design/frontend/Test/theme2/CP_Homepagereview/templates/homepersonalize.phtml
app/design/frontend/Test/theme2/CP_Homepagereview/web/js/breeze/personalization.js

ALso please find the code of phtml file where component is initialized.

```
       <section class="personalize-home deal" id="personalize-home" data-mage-init='{"CP_Homepagereview/js/personalization": {
                "currencySymbol": <?= json_encode($currencySymbol) ?>,
                "titleText": <?= json_encode($titleText) ?>,
                "productValues": <?= json_encode($productValues) ?>,
                "shippingPrice" : <?= json_encode($viewModel->getFreeShippingPrice()); ?>,
                "cartCouponText" : <?= json_encode($viewModel->getCouponDataCart()); ?>
            }}'>
            <div class="container">
                <div class="block-title title">
                    <h2 id="block-personalize-heading" class="block-personalize-heading" role="heading" aria-level="2"></h2>     
                </div>                
                <div id="cart-homepage" class="block-content content" aria-labelledby="block-personalize-heading"></div>
            </div>
        </section>
```

**Reply by vovayatsyuk** [reply]
`breeze_cms_index_index.xml` is the wrong file to register js files. Use the `breeze_default.xml` file when referencing `breeze.js` block.

See https://breezefront.com/docs/components#component-registration

**Reply by Paresh-Dhanani** [reply]
https://docs.swissuplabs.com/m2/extensions/breeze/devdocs/layout/

As per above we can use any xml file with breeze_ prefix. Can't we? Also i have some other component where i have added my code in below xml file and where i am also getting problem like sometime it is visible and sometime not. so it is correct file or not?

app/design/frontend/Test/theme1/Module_Name/layout/breeze_catalog_category_view.xml
app/design/frontend/Test/base/Module_Name/layout/breeze_catalog_product_view_type_customtype.xml

**Reply by vovayatsyuk** [reply, marked as answer]
You can. But when referencing breeze.js always do this from the breeze_default.xml file.

**Reply by Paresh-Dhanani** [reply]
It worked. I have updated the xml filename and now it is working fine.
Thanks for your help and quick support.
